embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/injection/StorageModule.kt

```diff
@@ -50,4 +56,10 @@ internal class StorageModuleImpl(
             coreModule.jsonSerializer
         )
     }
+
+    init {
+        workerThreadModule
+            .scheduledWorker(WorkerName.PERIODIC_CACHE)
```

Comment:
Looking at the description, `BACKGROUND_REGISTRATION` might be a better fit for this?

Reply:
I wanted to use that one but it says: 

> recurring tasks that don't use I/O or block for long periods of time.

and this reads from the disk. 

I guess it does fall into

> Used primarily to perform short-lived tasks that need to execute only once

Reply:
Serializing disk access probably isn't a bad idea. We may want to give it a different name so it's not so specific about session/BA payload saving, but yeah, this seems fine.